CHANGELOG.md

```diff
@@ -117,20 +353,204 @@ changes since the last release, see the [diff on GitHub][unreleased].
+  validate that the default value for a parameter was valid for the parameter's [type][p7-32].
+
+  <details><summary>Related work items</summary>
+
+  - Issues: _None_.
+  - PRs: [#364]
+
+  </details>
+
+- Added support for resources to send trace information to DSC during command execution. DSC
+  Resources can emit JSON objects to stderr. If the object has a property in the following list
+  with a string value, DSC interprets the emitted object as a message of the matching level:
+  `Error`, `Warning`, `Info`, `Debug`, `Trace`.
+
+  For example, DSC would interpret a resource emitting the following JSON to stderr as a warning:
+
+  ```json
+  {"Warning":"Unable to access remote store, using cached local package data only"}
+  ```
+
+  DSC emits these messages along with its own messages when the specified trace level for the
+  command is equal to or lower than the message's level.
+
+  For more information about trace levels, see the [--trace-level][p7-33] option for the
+  [dsc][cmd] root command.
+
+  <details><summary>Related work items</summary>
+
+  - Issues: [#89][#89]
+  - PRs: [#287][#287]
+
+  </details>
+
+- Added validation to ensure resources return data for their instances that is valid against their
+  own instance JSON schema. Prior to this release, the return data wasn't validated.
+
+  <details><summary>Related work items</summary>
+
+  - Issues: [#251][#251]
+  - PRs: [#362][#362]
+
+  </details>
+
+- Added multi-line progress bars for the `dsc resource list` command to provide feedback to
+  interactive users about the resource discovery process. Prior to this release, the command
+  executed silently.
+
+  <details><summary>Related work items</summary>
+
+  - Issues: _None_.
+  - PRs: [#323][#323]
+
+  </details>
+
+- Added functionality to insert metadata for adapter resources to indicate if the incoming data is
+  for a configuration instead of direct resource invocation. Prior to this release, adapters had no
+  way of discerning between a single-instance call for a configuration and a direct resource
+  invocation.
+
+  With this change, DSC inserts the following into the data object sent to the adapter during a
+  `dsc config` command:
+
+  ```json
+  "metadata": {
+    "Microsoft.DSC": {
+      "context": "Configuration"
+    }
+  }
+  ```
+
+  Adapters can then check whether this value is set in the input data and handle it as-needed.
+
+  <details><summary>Related work items</summary>
+
+  - Issues: [#253][#253]
+  - PRs: [#348][#348]
+
+  </details>
+
+- Added the `Microsoft.Windows/RebootPending` resource, which checks whether a Windows machine has
+  a pending reboot. It can only be used for assertions, not to enforce state.
+
+  <details><summary>Related work items</summary>
+
+  - Issues: [#333][#333]
+  - PRs: [#344][#344]
+
+  </details>
+
+- Added the `Microsoft.DSC.Transitional/RunCommandOnSet` resource, which runs a specified
```

Comment:
I know we don't want to promote use of this resource, but it might not be clear to novice users that this can be used to run a PowerShell script.